packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -96,29 +130,37 @@ function callMethodOnPrototype (variable, methodName, ...args) {
 }
 
 function iterateOn (fnName, variable, callbackCode) {
-  return `${isCollection(variable)}
-    ? Array.from(${variable}).${fnName}((it) => ${callbackCode})
-    : Object.entries(${variable}).${fnName}(([key, value]) => ${callbackCode})`
+  return `((val) => {
+    return ${isCollection('val')}
+      ? Array.from(val).${fnName}((it) => ${callbackCode})
+      : Object.entries(val).${fnName}(([key, value]) => ${callbackCode})
+  })(${variable})`
 }
 
 function isCollection (variable) {
   return `Array.isArray(${variable}) || ${variable} instanceof Set || ${variable} instanceof WeakSet`
 }
 
-function throwOnSideEffect (variable, keyOrIndex, allowSetAccess, allowMapAccess) {
+function accessProperty (variable, keyOrIndex, allowMapAccess) {
   return `(() => {
     if (${variable} instanceof Set || ${variable} instanceof WeakSet) {
-      ${allowSetAccess
-        ? `return ${variable}.has(${keyOrIndex}) ? ${keyOrIndex} : undefined`
-        : 'throw new Error(\'Aborting because accessing a Set or WeakSet is not allowed\')'}
+      throw new Error('Accessing a Set or WeakSet is not allowed')
     } else if (${variable} instanceof Map || ${variable} instanceof WeakMap) {
       ${allowMapAccess
-        ? `return ${variable}.get(${keyOrIndex})`
-        : 'throw new Error(\'Aborting because accessing a Map or WeakMap is not allowed\')'}
-    } else if (Object.getOwnPropertyDescriptor(${variable}, ${keyOrIndex})?.get === undefined) {
-      return ${variable}[${keyOrIndex}]
+        ? `return ${callMethodOnPrototype(variable, 'get', keyOrIndex)}`
+        : 'throw new Error(\'Accessing a Map or WeakMap is not allowed\')'}
+    } else {
+      return ${throwOnSideEffect(variable, keyOrIndex)}
+    }
+  })()`
+}
+
+function throwOnSideEffect (variable, propertyName) {
+  return `(() => {
+    if (Object.getOwnPropertyDescriptor(${variable}, ${propertyName})?.get === undefined) {
+      return ${variable}[${propertyName}]
     } else {
-      throw new Error('Aborting because of possible side effects')
+      throw new Error('Posibility of side effect')
```

Comment:
all those errors should be more helpful for the user to understand the problem.
something like `Possibility of side effect accessing when evaluate ${stringifyExpression(node)}`

However, we might want to address this on future PRs

Reply:
This only truly becomes relevant when implementing template strings, as any errors thrown in conditions will just treat the condition as "not met" - but the error itself will be swallowed.

I'll see if I can improve it in this PR, or I will defer it till I implement support for template strings.